web/src/notifications.js

```diff
@@ -403,6 +421,15 @@ export function should_send_desktop_notification(message) {
         return true;
     }
 
+    // Followed topic wildcard mentions
+    if (
+        message.mentioned &&
+        user_topics.is_topic_followed(message.stream_id, message.topic) &&
+        user_settings.enable_followed_topic_wildcard_mentions_notify
```

Comment:
> Would it be a good idea to merge this logic into the `stream_data.receives_notifications` function which is called above?

No, because it deals with prioritizing the stream-specific setting (if exists) over the global notification setting for a stream.
> if there is any scope to centralize the user-topic logic similar to the stream notifications logic.

I think it would make more sense to centralize it in a similar way, when we will have global and topic-specific setting kind of thing.

`stream_data.receives_notifications` has been used at 3 more places (not counting tests); but the intention at all the places is to find:
* is stream_specific and default setting dissimilar
* fetching the stream-specific setting over global

Reply:
Great, makes sense to have the topic code separate then :+1: